pysatMadrigal/tests/test_methods_general.py

```diff
@@ -230,28 +316,178 @@ def test_load_bad_coords(self):
+            self.data, self.meta = general.load([self.temp_file])
 
         # Test the loaded data
-        tst_lines = self.datalines.split('\n')
-        header = tst_lines[0].split()
-        values = tst_lines[1].split()
+        self.eval_data_and_metadata()
 
-        # Assert the data columns and number of values are the same
-        assert len(header) == len(data.columns)
-        assert len(data.index) == len(tst_lines) - 1
+        # Test the logging message
+        captured = caplog.text
+        assert captured.find("duplicated time indices") >= 0
+        return
 
-        # Test the values for the meta data name and the first data value
-        for i, col in enumerate(header):
-            assert str(data[col][0]).find(values[i]) == 0
-            assert meta[col, meta.labels.name] == col
+
+class TestNetCDFFiles(object):
+    """Tests for general methods with NetCDF files."""
+
+    def setup(self):
+        """Create a clean testing setup."""
+
+        # Create testing directory
+        self.data_path = tempfile.TemporaryDirectory()
+
+        # Initialize a test file name
+        self.temp_files = []
+        self.xarray_coords = [{('time',): ['time'],
+                               ('lat',): ['lat'],
+                               ('lon',): ['lon'],
+                               ('time', 'lat', 'lon'): ['value']}]
+
+        # Initialize the output
+        self.data = None
+        self.meta = None
+
+        return
+
+    def teardown(self):
+        """Clean up previous testing."""
+
+        # Remove the temporary directory and file
+        for tfile in self.temp_files:
+            if os.path.isfile(tfile):
+                os.remove(tfile)
+        self.data_path.cleanup()
+
+        del self.data_path, self.temp_files, self.xarray_coords, self.data
+        del self.meta
+        return
+
+    def write_temp_files(self, nfiles=0):
+        """Write data to a temporary file, zipping if desired.
+
+        Parameters
+        ----------
+        nfiles : int
+            Number of temporary NetCDF files to write (default=0)
+
+        """
+
+        for i in range(nfiles):
+            tfile = os.path.join(self.data_path.name,
+                                 "temp{:d}.netCDF4".format(i))
+
+            # Write a temporary file with data, based off of example:
+            # https://opensourceoptions.com/blog/
+            #    create-netcdf-files-with-python/
+            with nc.Dataset(tfile, "w", format="NETCDF4") as ds:
+                # Write the file attributes
+                if nfiles > 1:
+                    ds.catalog_text = "catalog text test"
+
+                # Write the file dimensions, using Madrigal defaults
+                ds.createDimension('timestamps', None)
+                ds.createDimension('lat', 10)
+                ds.createDimension('lon', 10)
+
+                # Write the data
+                dat_dict = {dkey: ds.createVariable(dkey, 'f4', (dkey,))
```

Comment:
I mean, I am also assigning the time value to be a single integer... so I think I'm giving the "time" more than enough bytes!

Reply:
I did miss you were assigning an int in. For this file it does have enough resolution for the input. My point was more general. If I see time going into a file, I throw bits at it. 

In the spirit of testing like the code is run I'd suggest using the same datatype the Madrigal uses, if they have a common one. While I don't think in this case it will reveal a bug, I've generally found that any areas I don't test end up having some bugs hidden in there.

Reply:
There are double, float, and integer time values.  Since we do actually test this with real files through integration tests, I am going to leave this as simple as possible to reduce runtime.